### PR TITLE
completed challenge 8 - CORE: PATCH /api/articles/:article_id

### PR DESCRIPTION
**Pull request by Dominic20227**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -21,3 +28,54 @@ describe("CORE: GET /api/topics", () => {
...
+            author: "icellusedkars",
+            body: "I was hungry.",
+            created_at: "2020-01-07T14:08:00.000Z",
+            votes: 1,
+            article_img_url:
+              "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+          });
+        });
+      });
+  });
```
Comment (popatre) [inline __tests__/app.test.js]: Need to return the single article updated - at the moment, looks like you're getting an array

File: __tests__/app.test.js [hunk 2]
```diff
@@ -21,3 +28,54 @@ describe("CORE: GET /api/topics", () => {
...
+    return request(app)
+      .patch("/api/articles/7")
+      .send(patchData)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("400 bad request");
+      });
+  });
+
+  it("400: responds with bad request, when rying to insert  numeric data type into out of range article_id", () => {
```
Comment (popatre) [inline __tests__/app.test.js]: 404

File: __tests__/app.test.js [hunk 3]
```diff
@@ -21,3 +28,54 @@ describe("CORE: GET /api/topics", () => {
...
+    const patchData = { inc_votes: 2 };
+
+    return request(app)
+      .patch("/api/articles/99")
+      .send(patchData)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("404 not found");
+      });
+  });
```
Comment (popatre) [inline __tests__/app.test.js:252]: Missing a 400 test -
Status 400: Bad request, invalid article_id
E.g. PATCH /api/articles/notanumber

**Review by popatre — CHANGES_REQUESTED** [review]
Two bits to change/add:

- [ ] Need to send back single article, not an array
- [ ] Missing a 400 test -
Status 400: Bad request, invalid article_id
E.g. PATCH /api/articles/notanumber

**Review by nc-jonathan-mcguire — APPROVED** [review]
Looks much better for ticket 8 👍

**Pull request merged**